pkg/shared/utils/sync/distributed_lock_mock.go

```diff
@@ -0,0 +1,35 @@
+package sync
+
+import (
+	"github.com/pkg/errors"
+	"gorm.io/gorm"
+)
+
+var _ DistributedLock = &distributedLockMock{}
+
+// It appears that go-mocket does not offer a straightforward method to intercept the Rollback call. To make it easily
+// interceptable, this mock substitutes the rollback call in the Unlock method with a DELETE.
+
+func newDistributedLockMock(db *gorm.DB, lockID string) DistributedLock {
+	dl := NewDistributedLock(db, lockID)
+	return &distributedLockMock{dl: dl, db: db, id: lockID}
+}
+
+type distributedLockMock struct {
+	dl DistributedLock
+	db *gorm.DB
+	id string
+}
+
+func (dl *distributedLockMock) Lock() error {
+	return dl.dl.Lock()
+}
+
+func (dl *distributedLockMock) Unlock() error {
+	dl.dl.Unlock()
```

Comment:
this is causing a linting issue https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/actions/runs/4566392609/jobs/8058829565